src/engine/strat_engine/metadata/static_header.rs

```diff
@@ -483,17 +501,20 @@ pub mod tests {
         fn test_ownership(ref sh in static_header_strategy()) {
             let buf_size = bytes!(static_header_size::STATIC_HEADER_SECTORS);
             let mut buf = Cursor::new(vec![0; buf_size]);
-            prop_assert!(StaticHeader::setup(&mut buf).unwrap().is_none());
+            let read_results = StaticHeader::read_sigblocks(&mut buf);
+            prop_assert!(StaticHeader::repair_sigblocks(&mut buf, read_results).unwrap().is_none());
 
             sh.write(&mut buf, MetadataLocation::Both).unwrap();
 
-            prop_assert!(StaticHeader::setup(&mut buf)
+            let read_results_2 = StaticHeader::read_sigblocks(&mut buf);
```

Comment:
@mulkieran Could you recommend a better way of doing this while still indicating a clear difference between these reads? Would you prefer simply shadowing these variables in the same scope?

Reply:
In ```test_ownership``` you have a good argument for not invoking ```repair_sigblocks``` at all. This test was never designed to test the repair behavior; there are other tests that do. Instead, it makes sense to assert a useful fact about the result of ```read_sigblocks```. That approach will resolve your question about this identifier, also.